apps/sched/clkinfo.js

```diff
@@ -123,7 +124,17 @@
           this.interval = undefined;
           this.switchTimeout = undefined;
         },
-        run: function() { }
+        run: function() {
+          if (a.date) return; // ignore events
+          a.on = !a.on;
+          if (a.on) {
+            // use sched to start the alarm,
+            // so it takes care of timers etc
+            alarm.setAlarm(a.id, a);
```

Comment:
I wonder if it's this line that removes all other alarms. On inspection no id seems to be set for alarms through the Alarms & Timers app so maybe they all get lumped together and overwritten?

Reply:
Yes this was exactly it, I was being a fool!